File: common/rpc/RpcEvent.cpp

#include "RpcEvent.h"


using namespace zonda::common;

RpcEvent::RpcEvent(IMsg* msg, RpcHead* head):
    m_msg(msg), m_rpc_head(head), m_req(NULL)
{
    m_msg_type = m_rpc_head->msg_type;
}

RpcEvent::RpcEvent(RpcHead* head):
    m_msg(NULL), m_rpc_head(head)
{
    m_msg_type = m_rpc_head->msg_type;
}


RpcEvent::~RpcEvent()
{
    //NOTE:msg will be deleted by the user
    
    if (m_msg_type < 0)
    {
        delete m_rpc_head;
    }

}

int RpcEvent::get_type() const
{
    return EventType::RPC_EVENT;
}

bool RpcEvent::is_req() const
{
    if (m_msg_type > 0) return true;
    else return false;
}

bool RpcEvent::is_res() const
{
    if (m_msg_type > 0) return false;
    else return true;

}

int RpcEvent::get_op_type() const
{
    if (m_msg_type > 0) return m_msg_type;
    else return (-1*m_msg_type);
}

  
IMsg* RpcEvent::get_msg() const
{
    return m_msg;
}

RpcHead* RpcEvent::get_rpc_head() const
{
    return m_rpc_head;
}

//!Only for res event
void RpcEvent::set_req(IMsg* req)
{
    m_req = req;
}
//!Only for res event
IMsg* RpcEvent::get_req()
{
    return m_req;
}

File: common_java/src/main/java/com/alibaba/platform/zonda/common/rpc/RpcEvent.java
package com.alibaba.platform.zonda.common.rpc;

import com.alibaba.platform.zonda.common.seda.*;

public class RpcEvent implements IEvent
{
    private IMsg m_msg;
    private RpcHead m_rpc_head;
    private int m_msg_type;
    private IMsg m_req;
    public RpcEvent(IMsg msg, RpcHead rh)
    {
        m_msg_type = msg.getMsgType();
        m_msg = msg;
        m_rpc_head = rh;
        m_req = null;
    }
    
    public RpcEvent(RpcHead rh)
    {
        m_msg_type = rh.msgType.get();
        m_msg = null;
        m_rpc_head = rh;
        m_req = null;
    }
    
    public int getType()
    {
        return EventType.RPC_EVENT;
    }
    
    public boolean isReq()
    {
        if (m_msg_type > 0) return true;
        return false;
    }
    
    public boolean isRes()
    {
        if (m_msg_type > 0) return false;
        return true;        
    }
    
    public int getOpType()
    {
        if (m_msg_type > 0) return m_msg_type;
        return 0-m_msg_type;
    }
    
    public IMsg getMsg()
    {
        return m_msg;
    }
    
    public RpcHead getRpcHead()
    {
        return m_rpc_head;
    }
    
    //!Only for res event
    public void setReq(IMsg req)
    {
        m_req = req;
    }
    
    //!Only for res event
    public IMsg getReq()
    {
        return m_req;
    }
}